docker-compose.yml

```diff
@@ -30,6 +30,10 @@ services:
         condition: service_healthy
       postgres:
         condition: service_started
+    healthcheck:
```

Comment:
The Dockerfile already defines a healthcheck, is it necessary to duplicate it here?

Reply:
I was tempted to remove the one from the Dockerfile itself, but as it might be used by the consumers I decided to leave as it is.
The one in the docker-compose, even though it's the same approach, helps to orchestrate the required services all at once and ensure those services are up and running before running the infra tests which were written in [BATS ](https://github.com/bats-core/bats-core). Otherwise, if the `opbeans-go` service takes a bit to startup and the BATS are executed straight away they will fail as it has happened recently. For instance, the [build 6](https://apm-ci.elastic.co/blue/organizations/jenkins/apm-agent-go%2Fopbeans-go-mbp/detail/master/6/pipeline#log-282) failed for that reason.

Reply:
> The one in the docker-compose, even though it's the same approach, helps to orchestrate the required services all at once and ensure those services are up and running before running the infra tests which were written in BATS .

Isn't that the reason for having the `wait` stanza, rather than the `healthcheck` attribute? What's the difference in behaviour if you added just `wait`, but didn't add `healthcheck`, to docker-compose.yml?

I don't want to block anything, just trying to understand why we need to duplicate it.

Reply:
`wait` is just the name of the service (https://github.com/elastic/opbeans-go/pull/30/files#diff-4e5e90c6228fd48698d074241c2ba760R153), nothing else. 

The workflow is:
- Start all the services which are required by `opbeans-go`
- `opbeans-go` service will wait for those service to either be healthy or started. https://github.com/elastic/opbeans-go/blob/522fe76058ec570f72fb7f1a3e087589f13964e2/docker-compose.yml#L24-L32
- `wait` service will wait for the `opbeans-go` service to be healthy, aka the `healthcheck` for the `opbeans-go` was a success.

Therefore, the `wait` service could be renamed `xyz` and it helps to orchestrate the `docker-compose up -d` in the manner that the `opbeans-go` is up and running. At the moment, the startup time might be faster locally than in CI as the resources allocated in the workers are a bit limited.

Please let me know if you need further clarifications

Reply:
> wait is just the name of the service

Gotcha.

> wait service will wait for the opbeans-go service to be healthy, aka the healthcheck for the opbeans-go was a success.

What I am missing is why we can't just rely on the Dockerfile's "HEALTHCHECK" line

Please go ahead and merge, but if you could clarify that for me afterwards that would be great. I've seen others do it too, and I'm just wondering what I'm missing.

Reply:
the `healthcheck` does not ensure you that  the last container you launch it is healthy when `docker-compose up -d` finish, when you add the `wait` container you are ensuring that the latest container is `healthy` because of the `wait` container depends-on the latest container, `wait` would not launch until the latest container is healthy. It is easy to check if you execute `docker-compose up -d && docker-compose ps -a` most of times the lastest container would be unhealthy without the `wait` container.

Reply:
I just run a few iterations and I'd rather remove it from the Dockerfile as both approaches are equally doing the same.  And let's avoid any redundancy for the shake of simplicity.